api/api_keys/user.py

```diff
@@ -0,0 +1,62 @@
+import typing
+
+from django.db.models import QuerySet
+
+from organisations.models import Organisation
+from permissions.permission_service import (
+    get_permitted_environments_for_master_api_key,
+    get_permitted_projects_for_master_api_key,
+    is_master_api_key_environment_admin,
+    is_master_api_key_project_admin,
+    master_api_key_has_organisation_permission,
+)
+
+from .models import MasterAPIKey
+
+if typing.TYPE_CHECKING:
+    from environments.models import Environment
+    from projects.models import Project
+
+
+class APIKeyUser:
```

Comment:
Can we extract a common Abstract class here to ensure that the same methods are implemented on both this class and the user class?